From 5868205e0c5e34b926c8ffb51a02306f26cabd30 Mon Sep 17 00:00:00 2001
From: Donald Carr <donald.carr@nokia.com>
Date: Tue, 14 Jun 2011 16:02:05 +0000
Subject: [PATCH 1/2] guard pointers in declarative engine

---
 src/declarative/qml/qdeclarativebinding.cpp |   10 +++++++++-
 1 files changed, 9 insertions(+), 1 deletions(-)

diff --git a/src/declarative/qml/qdeclarativebinding.cpp b/src/declarative/qml/qdeclarativebinding.cpp
index ca3bc37..b3ea92a 100644
--- a/src/declarative/qml/qdeclarativebinding.cpp
+++ b/src/declarative/qml/qdeclarativebinding.cpp
@@ -244,8 +244,16 @@ QDeclarativeBinding::createBinding(Identifier id, QObject *obj, QDeclarativeCont
         return 0;
 
     QDeclarativeContextData *ctxtdata = QDeclarativeContextData::get(ctxt);
+    
+    if(!ctxtdata)
+        return 0;
 
-    QDeclarativeEnginePrivate *engine = QDeclarativeEnginePrivate::get(qmlEngine(obj));
+    QDeclarativeEngine *pEngine = qmlEngine(obj);
+    
+    if(!pEngine)
+        return 0;
+    
+    QDeclarativeEnginePrivate *engine = QDeclarativeEnginePrivate::get(pEngine);
     QDeclarativeCompiledData *cdata = 0;
     QDeclarativeTypeData *typeData = 0;
     if (engine && ctxtdata && !ctxtdata->url.isEmpty()) {
-- 
1.7.5.4

