From b401a024545ebe35373b4bf039c6d825fb0a8dff Mon Sep 17 00:00:00 2001
From: Donald Carr <donald.carr@nokia.com>
Date: Sun, 19 Jun 2011 15:12:23 +0000
Subject: [PATCH 2/2] Fix qt font issue

---
 src/gui/painting/qpaintengineex.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/gui/painting/qpaintengineex.cpp b/src/gui/painting/qpaintengineex.cpp
index 8510416..5105d9a 100644
--- a/src/gui/painting/qpaintengineex.cpp
+++ b/src/gui/painting/qpaintengineex.cpp
@@ -1084,7 +1084,7 @@ void QPaintEngineEx::drawStaticTextItem(QStaticTextItem *staticTextItem)
             changedHints = true;
         }
 
-        fill(qtVectorPathForPath(path), staticTextItem->color);
+        fill(qtVectorPathForPath(path), s->pen.color());
 
         if (changedHints) {
             s->renderHints = oldHints;
-- 
1.7.5.4

